posthog/models/cohort/util.py

```diff
@@ -257,7 +257,11 @@ def insert_static_cohort(person_uuids: List[Optional[uuid.UUID]], cohort_id: int
 
 def recalculate_cohortpeople(cohort: Cohort, pending_version: int) -> Optional[int]:
 
-    cohort_filter, cohort_params = format_person_query(cohort, 0, custom_match_field="id", has_joined_person_props=True)
+    from posthog.queries.person_query import PersonQuery
+
+    cohort_filter, cohort_params = PersonQuery(
+        Filter(data={"properties": cohort.properties}, team=cohort.team), cohort.team.pk
```

Comment:
quick flyby: If you pass property as `{"type": "cohort", "id": cohort.pk}` then this should work, since that will go to `parse_prop_grouped_clauses` which will go to `cohort_query` which will take care of behavioural filters.